src/r-bridge/lang-4.x/ast/parser/xml/internal/xml-to-json.ts

```diff
@@ -8,8 +8,11 @@ import type { XmlBasedJson } from '../input-format'
  * @param config    - The configuration to use (i.e., what names should be used for the attributes, children, ...)
  * @param xmlString - The xml input to parse
  */
-export function xlm2jsonObject(config: XmlParserConfig, xmlString: string): Promise<XmlBasedJson> {
-	return xml2js.parseStringPromise(xmlString, {
+export function xlm2jsonObject(config: XmlParserConfig, xmlString: string): XmlBasedJson {
+	let result: XmlBasedJson = {}
+	xml2js.parseString(xmlString, {
+		// we want this to be strictly synchronous!
```

Comment:
maybe add the reason as well? so that the callback is synchronous? 

Yet we should maybe incorporate a test of transforming an artificailly huge file so that if this becomes problematic in the future... at least we have a chance of knowing that?
#newissuetime